watertap/unit_models/electrodialysis_1D.py

```diff
@@ -1908,33 +1947,46 @@ def eq_pressure_drop(self, t):
             else:
 
                 @self.Constraint(
+                    self.flow_channel_set,
+                    self.flowsheet().time,
+                    self.diluate.length_domain,
                     doc="To calculate friction factor",
                 )
-                def eq_friction_factor(self):
+                def eq_friction_factor(self, c, t, x):
                     if (
                         self.config.friction_factor_method
                         == FrictionFactorMethod.Gurreri
                     ):
                         return (
-                            self.friction_factor
-                            == 4 * 50.6 * self.spacer_porosity**-7.06 * self.N_Re**-1
+                            self.friction_factor[c, t, x]
+                            == 4
+                            * 50.6
+                            * self.spacer_porosity**-7.06
+                            * self.N_Re[c, t, x] ** -1
                         )
                     elif (
                         self.config.friction_factor_method
                         == FrictionFactorMethod.Kuroda
                     ):
                         return (
-                            self.friction_factor
-                            == 4 * 9.6 * self.spacer_porosity**-1 * self.N_Re**-0.5
+                            self.friction_factor[c, t, x]
+                            == 4
+                            * 9.6
+                            * self.spacer_porosity**-1
+                            * self.N_Re[c, t, x] ** -0.5
                         )
 
         @self.Constraint(
+            self.flow_channel_set,
             self.flowsheet().time,
             doc="To calculate total pressure drop over a stack",
         )
-        def eq_pressure_drop_total(self, t):
-            return (
-                self.pressure_drop_total[t] == self.pressure_drop[t] * self.cell_length
+        def eq_pressure_drop_total(self, c, t):
+            return self.pressure_drop_total[c, t] == sum(
+                self.pressure_dx[c, t, x]
+                * self.cell_length
+                / len(self.difference_elements)
+                for x in self.difference_elements
```

Comment:
Is there not an integration scheme in Pyomo?

In my experience simple integration schemes like this are prone numerical error. My suggestion would be a Gauss Quadrature (or trapezoidal rule). 
Though depending on the intended system it may not be a big difference in accuracy.

Reply:
I think this is taken directly from the RO 1D models currently in WaterTAP, so I wasn't reinventing the approach of our unit models here. I would have to double check.

Reply:
Ah, got it.

Following up on that. Is the expected variation in pressure profile more or less 'complex' than in RO 1D? I saw in another thread that one of the goals of this PR is to get accurate pressure drops. A better integration scheme could be critical to achieve it.

Reply:
@hunterbarber is correct. This aligns with what is done for the 1DRO model. From the code, it appears that the preexisting formulation assumed constant dP/dx across each element, whereas @hunterbarber 's changes allow for varying dP/dx across the discretized length.

I need to refresh my memory on what each PressureDropMethod config option corresponds to (i.e., experimental vs Darcy_Weisbach), but if my interpretation is correct about dP/dx assumed as constant/uniform across elements, then I would opt for keeping that preexisting formulation as its own PressureDropMethod option, and @hunterbarber 's revised formulation would be a separate option. For example, in the RO model, we have an option that assumed that dP/dX is fixed and constant, another that just takes a pressure drop across the whole module as the fixed variable, and another that does a more detailed calculation (like the one shown in this PR).

To avoid widespread test failures and fear of increasing model complexity, we could maintain the preexisting formulation for pressure drop calculations as the default config option, and users would have to manually set this more detailed option for pressure drop calculations.

Reply:
@johnson12742 I think if you wanted to explore alternative integration schemes to improve accuracy and discuss more with IDAES/Pyomo teams, that'd be fine---but I would think that would lead to a separate PR if anything.